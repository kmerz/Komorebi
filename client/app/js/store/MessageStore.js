/*jshint esversion: 6 */
import AppDispatcher from '../dispatcher/AppDispatcher';
import EventEmitter from 'events';
import assign from 'object-assign';

var snackbar = {
  open: false,
  message: ""
};

var MESSAGE_EVENT = 'message';

var MessageStore = assign({}, EventEmitter.prototype, {
  getSnackbarOpen: function () {
    return snackbar.open;
  },
  getMessage: function () {
    return snackbar.message;
  },
  emitChange: function() {
    this.emit(MESSAGE_EVENT);
  },
  addChangeListener: function(callback) {
    this.on(MESSAGE_EVENT, callback);
  },
  removeChangeListener: function(callback) {
    this.removeListener(MESSAGE_EVENT, callback);
  }
});

AppDispatcher.register(function(action) {
  switch(action.actionType) {
    case "SHOW_MESSAGE":
      snackbar.message = action.message;
      snackbar.open = true;
      MessageStore.emitChange();
      break;
    case "HIDE_MESSAGE":
      snackbar = {
        message: "",
        open: false
      };
      MessageStore.emitChange(MESSAGE_EVENT);
      break;
    default:
      break;
  }
});
module.exports = MessageStore;
