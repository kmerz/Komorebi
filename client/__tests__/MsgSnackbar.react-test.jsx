import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';

import MsgSnackbar from '../app/js/msg_snackbar';
import injectTapEventPlugin from 'react-tap-event-plugin';
import ReactTestUtils from 'react-addons-test-utils';
import ReactDOM from 'react-dom';
import MessageStore from '../app/js/store/MessageStore';
injectTapEventPlugin();

describe('<MessageSnackbar />', () => {
  MessageStore.__Rewire__("snackbar", {message: "Hello", open: true});

  it('Should render MsgSnackbar', () => {
    const wrapper = shallow(
      <MsgSnackbar />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

})
