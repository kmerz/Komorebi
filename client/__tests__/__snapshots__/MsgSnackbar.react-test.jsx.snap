exports[`<MessageSnackbar /> Should render MsgSnackbar 1`] = `
<div>
  <Snackbar
    autoHideDuration={2000}
    message="Hello"
    onRequestClose={[Function]}
    open={true} />
</div>
`;
